src/main/java/ru/mercuriev/Controller.java

```diff
@@ -17,7 +18,7 @@
     @RequestMapping(value = "/next", method = RequestMethod.POST)
     @ResponseBody
     public int[][] next(@RequestBody int[][] gen) throws IOException {
-        return engine.nextGeneration(gen);
+        return engine.getNextGeneration(gen);
```

Comment:
nextGeneration ih this case may seem more confusing, becaose it looks like getter. If the param will have name generation, than method can be renamed to next: return engine.next(generation);